controllers/endpoint-controllers.js

```diff
@@ -27,6 +27,15 @@ exports.getCommentsByArticleId = (request, response, next) => {
   const { article_id: articleId } = request.params;
   const promises = [selectCommentsByArticleId(articleId), checkArticleIdExists(articleId)];
   Promise.all(promises)
-  .then(([commentsData]) => response.status(200).send({ comments: commentsData }))
-  .catch(next)
+    .then(([commentsData]) => response.status(200).send({ comments: commentsData }))
+    .catch(next)
+}
+
+exports.postCommentByArticleId = (request, response, next) => {
+  const { body: comment } = request;
+  const { article_id: articleId } = request.params;
+  checkArticleIdExists(articleId)
```

Comment:
you don't need this function `checkArticleIdExists` because the article_id references a foreign key on the articles table - so if it doesn't match any psql will throw the error for you on a code of 23503 

Always make use of the psql errors where you can as it saves you from not having to write as much code